src/viam/sdk/components/encoder/client.hpp

```diff
@@ -21,7 +21,8 @@ namespace sdk {
 class EncoderClient : public Encoder {
    public:
     EncoderClient(std::string name, std::shared_ptr<grpc::Channel> channel);
-    position get_position(const AttributeMap& extra, position_type position_type) override;
+    position get_position(const AttributeMap& extra,
+                          position_type position_type = position_type::unspecified) override;
```

Comment:
I assume this is intentional?

Reply:
Yeah, we define the default behavior in the base `encoder` class but I found that the defaults didn't propagate unless specifically re-set.

Reply:
I like your proposed change but I don't want to get this too bogged down right now. Reverting this `encoder` change, creating another ticket to switch defaults to using the `template method`, and have removed defaults from all the new methods in this PR.